src/eiger_fastcs/eiger_controller.py

```diff
@@ -50,53 +50,127 @@ async def initialise(self) -> None:
         connection = HTTPConnection(
             self._ip_settings, headers={"Content-Type": "application/json"}
         )
-        detector_status = await connection.get("detector/api/1.8.0/status/keys")
-        requests = [
-            connection.get(f"detector/api/1.8.0/status/{item}")
-            for item in detector_status["value"]
-        ]
-        values = await asyncio.gather(*requests)
-
-        for i in range(len(detector_status["value"])):
-            # FastCS Types
-            match values[i]["value_type"]:
-                case "float":
-                    datatype = Float()
-                case "int":
-                    datatype = Int()
-                case "bool":
-                    datatype = Bool()
-                case "str" | "datetime" | "State":
-                    datatype = String()
-
-            # append the names of criteria to the values list
-            values[i] = {**{"name": detector_status["value"][i]}, **values[i]}
-
-            # Set Attributes
-            match values[i]["access_mode"]:
-                case "rw":
-                    setattr(
-                        self,
-                        values[i]["name"],
-                        AttrRW(
-                            datatype,
-                            handler=EigerHandler(
-                                f'detector/api/1.8.0/status/{values[i]["name"]}'
-                            ),
-                        ),
-                    )
-
-                case "r":
-                    setattr(
-                        self,
-                        values[i]["name"],
-                        AttrR(
-                            datatype,
-                            handler=EigerHandler(
-                                f'detector/api/1.8.0/status/{values[i]["name"]}'
-                            ),
-                        ),
-                    )
+        subsystems = ["detector", "stream", "monitor"]
+        parameters = ["status", "config"]
+        PVs: list[str] = []
+        PV_exc: list[str] = []
+
+        for index, subsystem in enumerate(subsystems):
```

Comment:
```suggestion
        attributes: Mapping[str, Attribute] = {}
        for index, subsystem in enumerate(subsystems):
            ...
```

I think this would be much neater if in this logic had some separation and deferred calling setattr until the very end:
1. Map parameter meta data from device to an Attribute using the access mode ("r", "w", "rw")
2. Add the variable to the attributes dictionary while handling duplicates
3. Call setattr(self, name, attribute) for (name, attribute) in attributes

Step one should be as minimal as possible and just set a local variable. Only in step 2 should it do the involved parts of checking for duplicate names and making them unique. It can move the existing parameter with something like
```
d[renamed] = d.pop(old)
```
and then add the new parameter with the subsystem included. The generation of the name with a subsystem should be a function that is shared for updating existing and adding new.

Then step 3 is very simple and could possibly be a function in the core of FastCS for other devices to share.